cfn-resources/stream-instance/mongodb-atlas-streaminstance.json

```diff
@@ -0,0 +1,245 @@
+      "description": "Configuration options for an Atlas Stream Processing Instance.",
+      "properties": {
+        "Tier": {
+          "type": "string",
+          "description": "Selected tier for the Stream Instance. Configures Memory / VCPU allowances.",
+          "enum": [
+            "SP30"
+          ],
+          "title": "Stream Instance Tier"
+        }
+      },
+      "additionalProperties": false
+    },
+    "StreamsKafkaAuthentication": {
+      "type": "object",
+      "description": "User credentials required to connect to a Kafka Cluster. Includes the authentication type, as well as the parameters for that authentication mode.",
+      "properties": {
+        "Mechanism": {
+          "type": "string",
+          "description": "Style of authentication. Can be one of PLAIN, SCRAM-256, or SCRAM-512."
+        },
+        "Password": {
+          "type": "string",
+          "format": "password",
+          "description": "Password of the account to connect to the Kafka cluster."
+        },
+        "Username": {
+          "type": "string",
+          "description": "Username of the account to connect to the Kafka cluster."
+        }
+      },
+      "additionalProperties": false
+    },
+    "DBRoleToExecute": {
+      "type": "object",
+      "description": "The name of a Built in or Custom DB Role to connect to an Atlas Cluster.",
+      "properties": {
+        "Role": {
+          "type": "string",
+          "description": "The name of the role to use. Can be a built in role or a custom role."
+        },
+        "Type": {
+          "type": "string",
+          "description": "Type of the DB role. Can be either BuiltIn or Custom.",
+          "enum": [
+            "BUILT_IN",
+            "CUSTOM"
+          ]
+        }
+      },
+      "additionalProperties": false
+    },
+    "StreamsKafkaSecurity": {
+      "type": "object",
+      "description": "Properties for the secure transport connection to Kafka. For SSL, this can include the trusted certificate to use.",
+      "properties": {
+        "BrokerPublicCertificate": {
+          "type": "string",
+          "description": "A trusted, public x509 certificate for connecting to Kafka over SSL."
+        },
+        "Protocol": {
+          "type": "string",
+          "description": "Describes the transport type. Can be either PLAINTEXT or SSL."
+        }
+      },
+      "additionalProperties": false
+    },
+    "StreamsConnection": {
+      "type": "object",
+      "description": "Settings that define a connection to an external data store.",
+      "properties": {
+        "Name": {
+          "type": "string",
+          "description": "Human-readable label that identifies the stream connection."
+        },
+        "Type": {
+          "type": "string",
+          "description": "Type of the connection. Can be either Cluster or Kafka.",
+          "enum": [
+            "Kafka",
+            "Cluster",
+            "Sample"
```

Comment:
I copied the StreamConnection spec from the [API spec](https://cloud.mongodb.com/api/openapi/spec/2.0) and `Sample` is there, you will find it if you look for `"enum" : [ "Kafka", "Cluster", "Sample" ]`. Seems like this comes from [MMS definition of StreamsConnectionView](https://github.com/10gen/mms/blob/1b310007a9c41a5901e8a077322a29bdecae0aee/server/src/main/com/xgen/cloud/nds/streams/_public/model/ui/StreamsConnectionView.java#L37)

Reply:
We verified with Atlas Streams team that `Sample` type is not supported via API to be created/managed. But since this will be read-only in your resource I think we should be good here. WDYT?

Reply:
Agreed, let's keep it so we don't have to change it in the future